tools/mo/openvino/tools/mo/front/common/partial_infer/utils.py

```diff
@@ -179,12 +179,19 @@ def int64_array(value: Union[Iterable[Union[float, int]], float, int]) -> np.nda
     return np.array(value, dtype=np.int64)
 
 
+def float32_array(value: Union[Iterable[Union[float, int]], float, int]) -> np.ndarray:
+    return np.array(value, dtype=np.float32)
+
+
 def float_array(value: Union[Iterable[Union[float, int]], float, int]) -> np.ndarray:
-    return np.array(value, dtype=np.float64)
+    return float32_array(value)
 
 
-def float32_array(value: Union[Iterable[Union[float, int]], float, int]) -> np.ndarray:
-    return np.array(value, dtype=np.float32)
+def mo_array(value: Union[Iterable[Union[float, int]], float, int], dtype=None) -> np.ndarray:
+    x = np.array(value, dtype=dtype)
+    if not isinstance(value, np.ndarray) and x.dtype == np.float64 and dtype != np.float64:
```

Comment:
not clear the logic here because `x.dtype == dtype` after line #191 and in the mean time there is expression `x.dtype == np.float64 and dtype != np.float64` how can it be? If it relates to `dtype==None` case, I propose to use `dtype is None`.

Also, please add description for this function. Not clear how it is used. What is a case when `not isinstance(value, np.ndarray)`?

Reply:
On line 191 if dtype is None the resulting dtype may be unexpected, so if it happened to become fp64 we force it to fp32. `not isinstance(value, np.ndarray)` is for the cases when we get actual np.ndarray in this function and we don't need to change the precision of it.